File: tools/misc/python/test-parameters.py
# TOOL test-parameters.py: "Test parameters in Python" ()
# INPUT file{...}.tsv: "Files to include" TYPE GENERIC
# OUTPUT output.txt: ""
# PARAMETER OPTIONAL int TYPE INTEGER FROM -100 TO 100 DEFAULT 0
# PARAMETER OPTIONAL dec TYPE DECIMAL FROM -0.1 TO 1.0 DEFAULT 0
# PARAMETER OPTIONAL string: String TYPE STRING
# PARAMETER OPTIONAL enum: Enum TYPE [first: First, second: Second, third: Third] DEFAULT first
# PARAMETER OPTIONAL column: Column TYPE COLUMN_SEL
# PARAMETER OPTIONAL metacolumn: "Phenodata column" TYPE METACOLUMN_SEL
# PARAMETER OPTIONAL unchecked: "Unchecked string" TYPE UNCHECKED_STRING

# deprecated
# PARAMETER OPTIONAL file: File TYPE INPUT_SEL

with open('output.txt', 'w') as f:
    f.write('int	' + str(int) + '\n')
    f.write('dec	' + str(dec) + '\n')
    f.write('string	' + string + '\n')
    f.write('enum	' + str(enum) + '\n')
    f.write('column	' + str(column) + '\n')
    f.write('metacolumn	' + str(metacolumn) + '\n')
#	f.write('string	' + unchecked + '\n')
#	f.write('file	' + str(file) + '\n')

File: tools/misc/R/test-parameters.R
# TOOL test-parameters.R: "Test parameters in R" (Tool for testing all parameter types)
# INPUT normalized.tsv: "Input file" TYPE GENERIC (Input file description.)
# INPUT OPTIONAL normalized2.tsv: "Another optional file" TYPE GENERIC (Another input file description.)
# OUTPUT output.tsv: "Output file"
# PARAMETER OPTIONAL int: Integer TYPE INTEGER FROM -100 TO 100 DEFAULT 0 (integer parameter)
# PARAMETER OPTIONAL dec: Decimal TYPE DECIMAL FROM -0.1 TO 1.0 DEFAULT 0 (decimal parameter)
# PARAMETER OPTIONAL string: String TYPE STRING (string parameter)
# PARAMETER OPTIONAL unchecked: "Unchecked string" TYPE UNCHECKED_STRING (string parameter with special characters)
# PARAMETER OPTIONAL enum: Enum TYPE [first: First, second: Second, third: Third] DEFAULT first (multiple selection)
# PARAMETER OPTIONAL column: Column TYPE COLUMN_SEL (select column from input file)
# PARAMETER OPTIONAL metacolumn: "Phenodata column" TYPE METACOLUMN_SEL (select column from phenodata)
# PARAMETER OPTIONAL file: File TYPE INPUT_SEL (select one of the inputs)

print(paste("int: ", int))
print(paste("dec: ", dec))
print(paste("string: ", string))
print(paste("unchecked:", unchecked))
print(paste("enum:", enum))
print(paste("column:", column))
print(paste("metacolumn:", metacolumn))
print(paste("file:", file))

system("mv normalized.tsv output.tsv")
